ITE/6/spec/provenance.md

```diff
@@ -0,0 +1,191 @@
+# Predicate type: Provenance v1
+
+Type URI: https://in-toto.io/Provenance/v1
+
+## Purpose
+
+Describe how an artifact or set of artifacts was produced.
+
+The primary focus is on automated builds that followed some "recipe".
+
+## Schema
+
+```jsonc
+{
+  "subject": { ... }
+  "predicateType": "https://in-toto.io/Provenance/v1",
+  "predicate": {                           // required
+    "builder": {                           // required
+      "id": "<URI>"                        // required
+    },
+    "recipe": {                            // optional
+      "type": "<URI>",                     // required
+      "definedInMaterial": /* integer */,  // optional
+      "entryPoint": "<STRING>",            // optional
+      "arguments": { /* object */ },       // optional
+      "reproducibility": { /* object */ }  // optional
+    },
+    "metadata": {                          // optional
+      "buildStartedOn": "<TIMESTAMP>",     // optional
+      "materialsComplete": true/false      // optional
+    },
+    "materials": [
+      {
+        "uri": "<URI>",                    // optional
+        "digest": { /* DigestSet */ },     // optional
+        "mediaType": "<MEDIA_TYPE>",       // optional
+        "tags": [ "<STRING>" ]             // optional
+      }
+    ]
+  }
+}
+```
+
+### Fields
+
+<a id="builder"></a>
+`builder` _object, required_
+
+> Idenfifies the entity that executed the build steps. Example:
+>
+> ```json
+> "builder": {
+>   "id": "https://github.com/Attestations/GitHubHostedActions@v1"
+> }
+> ```
+>
+> This is distinct from the signer because one signer may generate attestations
+> for more than one builder. For example, a single GitHubActions signer may
+> produce attestations for both "github-hosted runner" and various "self-hosted
+> runner" builders.
+>
+> Even though it may be implicit from the signer, it is required to aid
+> readability and debugging.
+>
+> Verifiers MUST only accept specific builders from specific signers.
+
+<a id="builder.id"></a>
+`builder.id` _string ([TypeURI]), required_
+
+> URI indicating the builder's identity.
+
+<a id="recipe"></a>
+`recipe` _object, optional_
+
+> Describes the actions that the builder performed. Example:
+>
+> ```json
+> "recipe": {
+>   "type": "https://github.com/Attestations/GitHubActionsWorkflow@v1",
+>   "definedInMaterial": 0,
+>   "entryPoint": "build.yaml:maketgz"
+> }
+> ```
+>
+> MAY be omitted if unknown or implicit from the builder.
+
+<a id="recipe.type"></a>
+`recipe.type` _string ([TypeURI]), required_
+
+> URI indicating what type of recipe was performed. It determines the meaning of
+> `recipe.entryPoint`, `recipe.arguments`, `recipe.reproducibility`, and
+> `materials`.
+
+<a id="recipe.definedInMaterial"></a>
+`recipe.definedInMaterial` _string, optional_
```

Comment:
This should be int, right? It's mentioned as an int above and in the example.

Reply:
Oops! Fixed.